File: src/main/java/com/example/TodoController.java
package com.example;

import java.util.*;
import java.util.Arrays;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import antlr.collections.List;

@RestController
public class TodoController {
	
	@Autowired
	private TodoService todoService;
	
	@RequestMapping(value="/todos")
	public java.util.List<Todo> index(){
		return todoService.getAll();
	}
	
	@RequestMapping(method=RequestMethod.POST, value="/todos")
	public Todo create(@RequestBody Todo todo){
		return todoService.addTodo(todo);
	}
	
	@RequestMapping(method=RequestMethod.PUT, value="/todos/{id}")
	public Todo update(@RequestBody Todo todo, @PathVariable Long id){
		todo.setId(id);
		return todoService.updateTodo(todo);
	}
	
	@RequestMapping(method=RequestMethod.DELETE, value="/todos/{id}")
	public void delete(@PathVariable Long id){
		Todo todo = todoService.findOne(id);
		todoService.deleteTodo(todo);
	}
}

File: src/main/resources/static/app/todo.controller.js
function TodoController($http) {
  var vm = this;
  
  vm.todos = [];
  vm.todo;

  $http.get('/todos').then(function(resp){
    resp.data.forEach(function(i){
      vm.todos.push(i)
    })
  })

  this.addTodo = function(){
    debugger
    $http({
      method: 'POST',
      url: '/todos',
      data: {'title': vm.todo}
    }).then(function(resp){
      debugger
      vm.todos.push(resp.data)
    })

    vm.todo = '';  
  }

  this.done = function(todo){
    var id = todo.id
      for(let i = 0; i < vm.todos.length; i++){
        if(vm.todos[i].id === id){
          var index = vm.todos.indexOf(vm.todos[i])
        }
      }

    todo.checked = true;

    $http({
      method: 'PUT',
      url: '/todos/' + todo.id,
      data: todo
    }).then(function(resp){
      vm.todos[index] = resp.data
    }) 
  }

  this.deleteTodo = function(todo){
    var id = todo.id
        for(let i = 0; i < vm.todos.length; i++){
          if(vm.todos[i].id === id){
            var index = vm.todos.indexOf(vm.todos[i])
          }
        }
        //debugger
    vm.todos.splice(index, 1)

    $http({
      method: 'DELETE',
      url: '/todos/' + todo.id,        
      data: todo
      }) 
  }

}
  
angular
  .module('app')
  .controller('TodoController', ['$http', TodoController]);